spec/rspec/core/example_spec.rb

```diff
@@ -193,6 +193,7 @@ def assert(val)
         expect(RSpec::Matchers).not_to receive(:generated_description)
         example_group.example { assert 5 == 5 }
         example_group.run
+        RSpec::Mocks.space.reset_all
```

Comment:
Hmm, I wonder if we should revert #1862 instead?

BTW, what failure do you get w/o this line?